perceval/utils/parameter.py

```diff
@@ -212,16 +218,197 @@ def max(self, m: bool):
+        elif isinstance(other, Parameter):
+            return Expression(f"({self.name}*{other.name})", {self, other})
+        elif isinstance(other, (int, float)):
+            return Expression(f"({other}*{self.name})", {self})
+        raise TypeError("Unsupported parameter operation.")
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __add__(self, other):
+        if isinstance(other, Expression):
+            return Expression(f"({self.name}+{other.name})", {self} | other.params)
+        elif isinstance(other, Parameter):
+            return Expression(f"({self.name}+{other.name})", {self, other})
+        elif isinstance(other, (int, float)):
+            return Expression(f"({self.name}+{other})", {self})
+        raise TypeError("Unsupported parameter operation.")
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        if isinstance(other, Expression):
+            return Expression(f"({self.name}-{other.formula})", {self} | other.params)
+        elif isinstance(other, Parameter):
+            return Expression(f"({self.name}-{other.name})", {self, other})
+        elif isinstance(other, (int, float)):
+            return Expression(f"({self.name}-{other})", {self})
+        raise TypeError("Unsupported parameter operation.")
+
+    def __rsub__(self, other):
+        if isinstance(other, (int, float)):
+            return Expression(f"({other}-{self.name})", {self})
+        raise TypeError("Unsupported operation.")
+    
+    def __truediv__(self, other):
+        if isinstance(other, Expression):
+            return Expression(f"({self.name}/{other.name})", {self} | other.params)
+        elif isinstance(other, Parameter):
+            return Expression(f"({self.name}/{other.name})", {self, other})
+        elif isinstance(other, (int, float)):
+            return Expression(f"({self.name}/{other})", {self})
+        raise TypeError("Unsupported parameter operation.")
+    
+    def __pow__(self, other):
+        if isinstance(other, Expression):
+            return Expression(f"({self.name}^{other.name})", {self} | other.params)
+        elif isinstance(other, Parameter):
+            return Expression(f"({self.name}^{other.name})", {self, other})
+        elif isinstance(other, (int, float)):
+            return Expression(f"({self.name}^{other})", {self})
+        raise TypeError("Unsupported parameter operation.")
+    
+    def __neg__(self):
+        # Ensure using __neg__ twice in a row returns the original parameter
+        if self._original is not None:
+            return self._original
+        
+        if isinstance(self, Expression):
+            expr = Expression(f"(-{self.name})", self._params)
+        else:
+            expr = Expression(f"(-{self.name})", {self})
+        expr._original = self
+        return expr
 
 class Expression(Parameter):
```

Comment:
I suggest we move this class to a new expression.py file, for maintainability reasons.